client/rpc-spec-v2/Cargo.toml

```diff
@@ -36,6 +36,7 @@ tokio = { version = "1.22.0", features = ["sync"] }
 array-bytes = "6.1"
 log = "0.4.17"
 futures-util = { version = "0.3.19", default-features = false }
+async-channel = "1.8.0"
```

Comment:
Can you explain why `async-channel` is used here? 

You already got the `futures-channel and tokio sync` in the dependency tree so I don't follow why this dependency is introduced as the both tokio and futures-channel provides an API for the stuff you implemented on top of the channel

Reply:
That makes sense! I can't actually remember if I picked the `async-channel` here since it doesn't require the receiver part to be mutable 🤔 

I'll change a bit the API since it's worth not adding extra dependencies, thanks!

Reply:
ah, didn't notice that :D